audit/internal2/README.md

```diff
@@ -0,0 +1,204 @@
+# Internal audit of autonolas-staking-programmes
+The review has been performed based on the contract code in the following repository:<br>
+`https://github.com/valory-xyz/ai-registry-mech` <br>
+commit: 7a8708d345a2201a1b693164f97dd9266532ea1b (tag: v0.4.0-pre-internal-audit) <br> 
+
+## Objectives
+The audit focused on updated marketplace contracts in this repo. <br>
+Limits: The subject of the audit is not contracts used as library contracts. Thus, this audit is not a full-fledged audit of contracts underlying the contract ERC721Mech. <br>
+
+### Flatten version
+Flatten version of contracts. [contracts](https://github.com/valory-xyz/ai-registry-mech/blob/main/audits/internal1/analysis/contracts)
+
+### Security issues. Updated 17-12-2024
+#### Problems found instrumentally
+Several checks are obtained automatically. They are commented. <br>
+All automatic warnings are listed in the following file, concerns of which we address in more detail below: <br>
+[slither-full](https://github.com/valory-xyz/autonolas-staking-programmes/blob/main/audits/internal2/analysis/slither_full.txt) 
+Most of the issues raised by instrumental analysis are outside the scope of the audit. <br>
+
+
+### Issue
+#### Critical? _preDeliver() passed low then price and zero balance 
+```
+        // Check for the number of credits available in the subscription
```

Comment:
_preDeliver() should probably be reverted whenever creditsToBurn < creditsBalance. 
This means in particular that when there is not enough credit in the balance, the mech can't deliver (I think it would not want to). Probably need to change the Karma mechanism in case using subscription actually (cause the mech does not deliver because of low balance, not its fault).